test/e2e/container_iface_name_test.go

```diff
@@ -0,0 +1,277 @@
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func createNetworkDevice(name string) {
+	session := SystemExec("ip", []string{"link", "add", name, "type", "bridge"})
+	session.WaitWithDefaultTimeout()
+	Expect(session).Should(ExitCleanly())
+}
+
+func deleteNetworkDevice(name string) {
+	session := SystemExec("ip", []string{"link", "delete", name})
+	session.WaitWithDefaultTimeout()
+	Expect(session).Should(ExitCleanly())
+}
+
+func createContainersConfFileWithDeviceIfaceName(pTest *PodmanTestIntegration) {
+	configPath := filepath.Join(pTest.TempDir, "containers.conf")
+	containersConf := []byte("[containers]\ninterface_name = \"device\"\n")
+	err := os.WriteFile(configPath, containersConf, os.ModePerm)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Set custom containers.conf file
+	os.Setenv("CONTAINERS_CONF_OVERRIDE", configPath)
+	if IsRemote() {
+		pTest.RestartRemoteService()
+	}
+}
+
+func isNetavark(p *PodmanTestIntegration) bool {
+	if p.NetworkBackend == Netavark {
+		GinkgoWriter.Println("This test is not compatible with the netavark network backend")
+		return true
+	}
+
+	return false
+}
+
+var _ = Describe("Podman container interface name", func() {
+
+	It("podman container interface name for bridge network", func() {
+		// Assert that the network interface name inside container for
+		// bridge network is ethX regardless of interface_name setting
+		// in the containers.conf file.
+
+		netName1 := createNetworkName("bridge")
+		netName2 := createNetworkName("bridge")
+
+		defer podmanTest.removeNetwork(netName1)
+		nc1 := podmanTest.Podman([]string{"network", "create", netName1})
+		nc1.WaitWithDefaultTimeout()
+		Expect(nc1).Should(ExitCleanly())
+
+		defer podmanTest.removeNetwork(netName2)
+		nc2 := podmanTest.Podman([]string{"network", "create", netName2})
+		nc2.WaitWithDefaultTimeout()
+		Expect(nc2).Should(ExitCleanly())
+
+		for _, override := range []bool{false, true} {
+			if override {
+				createContainersConfFileWithDeviceIfaceName(podmanTest)
+			}
+
+			ctr := podmanTest.Podman([]string{"run", "-d", "--network", netName1, "--name", "test", ALPINE, "top"})
+			ctr.WaitWithDefaultTimeout()
+			Expect(ctr).Should(ExitCleanly())
+
+			exec1 := podmanTest.Podman([]string{"exec", "test", "ip", "addr", "show", "eth0"})
+			exec1.WaitWithDefaultTimeout()
+			Expect(exec1).Should(ExitCleanly())
+			Expect(exec1.OutputToString()).Should(ContainSubstring("eth0"))
+
+			conn := podmanTest.Podman([]string{"network", "connect", netName2, "test"})
+			conn.WaitWithDefaultTimeout()
+			Expect(conn).Should(ExitCleanly())
+
+			exec2 := podmanTest.Podman([]string{"exec", "test", "ip", "addr", "show", "eth1"})
+			exec2.WaitWithDefaultTimeout()
+			Expect(exec2).Should(ExitCleanly())
+			Expect(exec2.OutputToString()).Should(ContainSubstring("eth1"))
+
+			rm := podmanTest.Podman([]string{"rm", "--time=0", "-f", "test"})
+			rm.WaitWithDefaultTimeout()
+			Expect(rm).Should(ExitCleanly())
```

Comment:
please remove this part from all your tests, they will clean up on their own and this juts bloats the lines with unnecessary code.

Reply:
I had removed those statements but the test failed when it went in the next iteration that the container with the same name was already running. Definitely it didn’t auto cleanup at the end of the iteration. So, I had to add them back.